internal/route/api/v1/repo/commits.go

```diff
@@ -18,6 +19,52 @@ import (
 	"gogs.io/gogs/internal/gitutil"
 )
 
+// GetAllCommits returns a slice of commits starting from HEAD.
+func GetAllCommits(c *context.APIContext) {
+	if strings.Contains(c.Req.Header.Get("Accept"), api.MediaApplicationSHA) {
+		c.SetParams("*", c.Params(":sha"))
+		GetReferenceSHA(c)
+		return
+	}
+
+	// Get pagesize, set default if it is not specified.
+	pageSizeStr := c.Query("pageSize")
+	if pageSizeStr == "" {
+		pageSizeStr = "30"
+	}
+
+	pageSize, err := strconv.Atoi(pageSizeStr)
+	if err != nil {
+		c.Error(err, "get page size error")
+	}
+
+	gitRepo, err := git.Open(c.Repo.Repository.RepoPath())
+	if err != nil {
+		c.Error(err, "open repository")
+		return
+	}
+
+	// The response object returned as JSON
+	result := []*api.Commit{}
```

Comment:
```suggestion
	result := make([]*api.Commit, 0, pageSize)
```

We can preallocate to the `pageSize`.